src/main/java/com/synopsys/integration/detect/workflow/bdio/CodeLocationBdioCreator.java

```diff
@@ -51,11 +69,32 @@ public CodeLocationBdioCreator(final DetectBdioWriter detectBdioWriter, final Si
             final ExternalId externalId = bdioCodeLocation.getDetectCodeLocation().getExternalId();
             final DependencyGraph dependencyGraph = bdioCodeLocation.getDetectCodeLocation().getDependencyGraph();
 
+            // Bdio 1
             final SimpleBdioDocument simpleBdioDocument = simpleBdioFactory.createSimpleBdioDocument(codeLocationName, projectNameVersion.getName(), projectNameVersion.getVersion(), externalId, dependencyGraph);
 
-            final File outputFile = new File(bdioOutput, bdioCodeLocation.getBdioName());
-            detectBdioWriter.writeBdioFile(outputFile, simpleBdioDocument);
-            uploadTargets.add(UploadTarget.createDefault(codeLocationName, outputFile));
+            final File bdio1OutputFile = new File(bdioOutput, bdioCodeLocation.getBdioName() + ".jsonld");
+            detectBdioWriter.writeBdioFile(bdio1OutputFile, simpleBdioDocument);
+            uploadTargets.add(UploadTarget.createDefault(codeLocationName, bdio1OutputFile));
+
+            // Bdio 2
+            final ProductList.Builder productListBuilder = new ProductList.Builder();
+            final String detectVersion = detectInfo.getDetectVersion();
+            final SpdxCreator detectCreator = SpdxCreator.createToolSpdxCreator("Detect", detectVersion);
+            final Product product = new Product.Builder().name(detectCreator.getIdentifier()).build();
+            productListBuilder.addProduct(product);
+
+            final BdioMetadata bdioMetadata = bdio2Factory.createBdioMetadata(codeLocationName, ZonedDateTime.now(), productListBuilder);
+            final Project bdio2Project = bdio2Factory.createProject(externalId, projectNameVersion.getName(), projectNameVersion.getVersion());
+            final Bdio2Document bdio2Document = bdio2Factory.createBdio2Document(bdioMetadata, bdio2Project, dependencyGraph);
+
+            final Bdio2Writer bdio2Writer = new Bdio2Writer();
+            final File bdio2OutputFile = new File(bdioOutput, bdioCodeLocation.getBdioName() + ".bdio");
+            try {
+                final OutputStream outputStream = new FileOutputStream(bdio2OutputFile);
+                bdio2Writer.writeBdioDocument(outputStream, bdio2Document);
```

Comment:
Is this file an archive of multiple files (including the header and one or more entries)?   Is chunking supported?

Reply:
1) Not for 6.1.0. That is coming later.
2) Yes it is an archive with the .bdio file extension.
3) Yes and results were the same.
4) For 6.1.0 Detect can only generate bdio in offline mode, and in that mode files from the run are preserved on the disk. When we support bdio 2 in online mode the user can also set detect.cleanup=false to have the files persist.

Reply:
Thanks.  For SCA on Polaris, the intent will be for the Polaris CLI to drive Detect dependency scans with output of the BDIO2 header and chunked entry files.  For standalone Black Duck, the system primarily has accepted start mode header, multiple chunk additions, and a completion message.   The archive has primarily been for generic download/upload via the UI, but we don't have any clients using this mechanism right now.

Reply:
The BDIO library (https://github.com/blackducksoftware/bdio) currently has no implementation of a StreamSupplier other than a BdioFile (https://github.com/blackducksoftware/bdio/blob/1cd704a24ee2affbf03b635e3e41d5be8b2fd329/bdio2/src/main/java/com/blackducksoftware/bdio2/BdioWriter.java#L74). If another implementation of StreamSupplier was implemented that only outputs the header and chunked entries instead of an archive, we could use that.

Reply:
@JakeMathews @ketul-shukla, can you get together with Jake to better understand this mechanism?  Currently, we are definitely chunking BDIO2 files internally.  What is the library mechanism to support this type of structure?